core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -662,6 +663,22 @@ get_sol_socket_option(int sockfd, int optname, void *__restrict optval,
             error = __wasi_sock_get_broadcast(sockfd, (bool *)optval);
             HANDLE_ERROR(error);
             return error;
+        case SO_TYPE:
+            assert(*optlen == sizeof(int));
+            error = __wasi_fd_fdstat_get(sockfd, &sb);
+            HANDLE_ERROR(error);
+            switch (sb.fs_filetype) {
+                case __WASI_FILETYPE_SOCKET_DGRAM:
+                    *(int *)optval = SOCK_DGRAM;
+                    break;
+                case __WASI_FILETYPE_SOCKET_STREAM:
+                    *(int *)optval = SOCK_STREAM;
+                    break;
+                default:
+                    errno = ENOPROTOOPT;
+                    return -1;
+            }
+            return 0;
```

Comment:
```suggestion
            return __WASI_ERRNO_SUCCESS;
```

Reply:
... `getsockopt()`? I'm not sure why it's mentioned here, but if using it as an example, it actually returns `-1` or `__WASI_ERRNO_SUCCESS`, while `__WASI_ERRNO_SUCCESS` has a value of 0. 

Returning `__WASI_ERRNO_SUCCESS` instead of 0 appears to be a common practice in other functions within wasi_socket_ext.c.

Reply:
> ... `getsockopt()`? I'm not sure why it's mentioned here, but if using it as an example, it actually returns `-1` or `__WASI_ERRNO_SUCCESS`, while `__WASI_ERRNO_SUCCESS` has a value of 0.

this function is a part of getsockopt.
 
> Returning `__WASI_ERRNO_SUCCESS` instead of 0 appears to be a common practice in other functions within wasi_socket_ext.c.

is it a common practice? yes.
is it good? no. it's wrong.

Reply:
So, should we create another PR to address everything else while maintaining common practice in this PR?